app/src/main/java/ch/epfl/favo/user/UserUtil.java

```diff
@@ -22,7 +22,7 @@
   TODO: Design singleton constructor and logic
    */
   // Expose current logged in user id
-  public static String currentUserId = "currentUserId";
+  public static String currentUserId = DependencyFactory.getCurrentFirebaseUser().getUid();
```

Comment:
I don't think this is correct: the user id is not a constant and should not be a static field in my opinion because this is initialized at the beginning and never updated, while the user id can change when a user logs in or logs out. I would just remove this field and call DependencyFactory.getCurrentFirebaseUser().getUid() directly whenever it's needed since the FirebaseUser class takes care to update the id during the authentication flow.

Reply:
I agree with you, I didn't think of that making the change now